cibuildwheel/windows.py

```diff
@@ -1,91 +1,132 @@
 from collections import namedtuple
+from contextlib import contextmanager
 from glob import glob
 
 try:
     from shlex import quote as shlex_quote
 except ImportError:
     from pipes import quote as shlex_quote
 
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
 from .util import prepare_command, get_build_verbosity_extra_flags
+if hasattr(sys, 'getwindowsversion'):
+    from ._vendored.pep514tools import findone as _pep514_findone
 
 
 IS_RUNNING_ON_AZURE = os.path.exists('C:\\hostedtoolcache')
 IS_RUNNING_ON_TRAVIS = os.environ.get('TRAVIS_OS_NAME') == 'windows'
+IS_RUNNING_ON_APPVEYOR = os.environ.get('APPVEYOR', 'false').lower() == 'true'
+
+
+@contextmanager
+def _mkdtemp():
+    path = tempfile.mkdtemp()
+    try:
+        yield path
+    finally:
+        shutil.rmtree(path)
+
 
 def get_python_path(config):
-    if IS_RUNNING_ON_AZURE:
-        # We can't hard-code the paths because on Azure, we don't know which
-        # bugfix release of Python we are getting so we need to check which
-        # ones exist. We just use the first one that is found since there should
-        # only be one.
-        path_pattern = 'C:\\hostedtoolcache\\windows\\Python\\{version}\\{arch}'.format(
-            version=config.version.replace('x', '*'),
-            arch='x86' if config.arch == '32' else 'x64'
-        )
-        try:
-            return glob(path_pattern)[0]
-        except IndexError:
-            raise Exception('Could not find a Python install at ' + path_pattern)
-    elif IS_RUNNING_ON_TRAVIS:
-        if config.version == "3.4.x":
-            return config.path
-        else:
-            nuget_args = get_nuget_args(config)
-            return os.path.join(nuget_args[-1], nuget_args[0] + "." + config.nuget_version, "tools")
+    major, minor = config.version.split('.')[:2]
+
+    if (int(major), int(minor)) < (3, 5):
+        if IS_RUNNING_ON_AZURE:
+            # We can't hard-code the paths because on Azure, we don't know which
+            # bugfix release of Python we are getting so we need to check which
+            # ones exist. We just use the first one that is found since there should
+            # only be one.
+            path_pattern = 'C:\\hostedtoolcache\\windows\\Python\\{version}\\{arch}'.format(
+                version=config.version.replace('x', '*'),
+                arch='x86' if config.arch == '32' else 'x64'
+            )
+            matches = glob(path_pattern)
+            if len(matches) > 0:
+                return matches[0]
+        elif IS_RUNNING_ON_APPVEYOR:
+            # We're running on AppVeyor
+            major, minor = config.version.split('.')[:2]
+            python_path = 'C:\\Python{major}{minor}{arch}'.format(
+                major=major,
+                minor=minor,
+                arch = '-x64' if config.arch == '64' else ''
+            )
+            if os.path.exists(python_path):
+                return python_path
```

Comment:
If we have this code anyway, isn't it easier/slighlty simpler code to just look for known locations, even on Python 3.5?

I mean:
```
if IS_RUNNING_ON_AZURE:
    # ...
elif IS_RUNNING_ON_APPVEYOR:
    # ...

if (int(major), int(minor)) >= (3, 5):
    # PEP 514

# ...
```

Reply:
It's for the sake of consistency on all CI. Starting with python 3.5, always use `pep514tools` in order to find an official install. The one on Azure is not official (at least per pep514). I guess in less than a year, it won't matter anyway because Python 2.7 & 3.4 will be dropped thus leaving only a clean `pep514tools` method.

Reply:
Hmmmm, ok, yes, why not, then.